### PR TITLE
soc: npcx: Add output npcx image in hex format

### PR DESCRIPTION
**Pull request by WealianLiao**
To support west OpenOCD path image flashing, this generates an npcx hex
image from the npcx BIN image. Moreover, this also adds a config for
the name of the output image.

The OpenOCD side is under code review.
For testing, this can build by http://openocd.zylin.com/#/c/5950/
1. copy  `${OpenOCD}/src/openocd` to `{ZEPHYR_SDK_INSTALL_DIR}/sysroots/x86_64-pokysdk-linux/usr/bin/`
2. copy `${OpenOCD}/tcl/target/npcx.cfg` to `{ZEPHYR_SDK_INSTALL_DIR}/sysroots/x86_64-pokysdk-linux/usr/share/openocd/scripts/target/`

Then, NPCX evb can flash image by:
```
west flash --hex-file build/zephyr/zephyr.npcx.hex
```

### CODE REVIEW

File: soc/arm/nuvoton_npcx/Kconfig [hunk 1]
```diff
@@ -20,6 +20,27 @@ menuconfig NPCX_HEADER
...
+	help
+	  Build a "raw" binary zephyr/npcx_image.bin in the build directory.
+	  The name of this file can be customized with CONFIG_NPCX_IMAGE_NAME.
+
+config NPCX_IMAGE_OUTPUT_HEX
+	bool "Build npcx binary in HEX format"
+	depends on NPCX_IMAGE_OUTPUT_BIN
+	help
+	  Build an HEX binary zephyr/npcx_image.hex in the build directory.
+	  This generates from npcx BIN image.
```
Comment (sjg20) [inline soc/arm/nuvoton_npcx/Kconfig:41]: This is generated from the npcx BIN image

File: soc/arm/nuvoton_npcx/Kconfig [hunk 2]
```diff
@@ -20,6 +20,27 @@ menuconfig NPCX_HEADER
 
 if NPCX_HEADER
 
+config NPCX_IMAGE_NAME
```
Comment (sjg20) [inline soc/arm/nuvoton_npcx/Kconfig:23]: Is this necessary, or should we always just use the same name?

**Review by sjg20 — CHANGES_REQUESTED** [review]

File: soc/arm/nuvoton_npcx/Kconfig (hunk 2 above)
Comment (keith-zephyr) [reply soc/arm/nuvoton_npcx/Kconfig:23]: This name is used to create the output file generated by ecst.py.  It should be different from ${KERNEL_BIN_NAME}, which defaults to zephyr.bin.  
What about setting up a default extension instead? Something like ${CONFIG_KERNEL_BIN_NAME}.npcx.bin.

File: soc/arm/nuvoton_npcx/Kconfig (hunk 1 above)
Comment (WealianLiao) [reply soc/arm/nuvoton_npcx/Kconfig]: Done.

File: soc/arm/nuvoton_npcx/Kconfig (hunk 2 above)
Comment (WealianLiao) [reply soc/arm/nuvoton_npcx/Kconfig]: I'm OK to use CONFIG_KERNEL_BIN_NAME. Please also check the modified CL for chromium ec project: 'https://crrev.com/c/2909957'.

**Review by sjg20 — APPROVED** [review]

**Review by keith-zephyr — APPROVED** [review]

File: soc/arm/nuvoton_npcx/Kconfig [hunk 3]
```diff
@@ -20,6 +20,21 @@ menuconfig NPCX_HEADER
 
 if NPCX_HEADER
 
+config NPCX_IMAGE_OUTPUT_BIN
```
Comment (nashif) [inline soc/arm/nuvoton_npcx/Kconfig:23]: why don't you just use the generic option BUILD_OUTPUT_BIN defined in Kconfig.zephyr?

File: soc/arm/nuvoton_npcx/Kconfig [hunk 4]
```diff
@@ -20,6 +20,21 @@ menuconfig NPCX_HEADER
...
+	bool "Build npcx binary in BIN format"
+	default y
+	help
+	  Build a "raw" binary zephyr/zephyr.npcx.bin in the build directory.
+	  The name of this file can be customized with CONFIG_KERNEL_BIN_NAME.
+
+config NPCX_IMAGE_OUTPUT_HEX
```
Comment (nashif) [inline soc/arm/nuvoton_npcx/Kconfig:30]: why don't you just use the generic option BUILD_OUTPUT_HEX defined in Kconfig.zephyr?

File: soc/arm/nuvoton_npcx/Kconfig (hunk 3 above)
Comment (WealianLiao) [reply soc/arm/nuvoton_npcx/Kconfig:23]: The npcx hex binary is generated from an npcx raw binary. I would like to add a dependency checking in Kconfig, so I add an option here. Moreover, currently, some applications that image need firmware header, and some applications don't need the firmware header. I think adding options will more clear.

File: soc/arm/nuvoton_npcx/Kconfig (hunk 4 above)
Comment (WealianLiao) [reply soc/arm/nuvoton_npcx/Kconfig:30]: Please check the previous one.

**Comment by carlescufi** [comment]
@nashif please revisit

**Comment by WealianLiao** [comment]
PR #35440 is merged. I will rebase it & modify the npcx9evb.

**Review by nashif — APPROVED** [review]

**Pull request merged**